controllers/space/space_controller.go

```diff
@@ -387,6 +386,15 @@ func NewNSTemplateSet(namespace string, space *toolchainv1alpha1.Space, bindings
 		},
 	}
 	nsTmplSet.Spec = NewNSTemplateSetSpec(space, bindings, tmplTier)
+
+	// propagate feature annotation from the space
+	winners, found := space.Annotations[toolchainv1alpha1.FeatureToggleNameAnnotationKey]
+	if found {
+		if nsTmplSet.Annotations == nil {
+			nsTmplSet.Annotations = make(map[string]string)
+		}
+		nsTmplSet.Annotations[toolchainv1alpha1.FeatureToggleNameAnnotationKey] = winners
+	}
```

Comment:
This will work only for the new NSTemplateSets, not for the already existing ones. If I want to enable/disable the feature toggle for some Space, then this should get propagated as well, but since we compare only 
https://github.com/codeready-toolchain/host-operator/blob/012f0cdf58e4fe53e37a2122ef5eb176009527ad/controllers/space/space_controller.go#L335-L336
then the annotation change won't get there.
Let's also make sure that it's tested in unit tests

Reply:
Great catch! Addressed in https://github.com/codeready-toolchain/host-operator/pull/1048/commits/601661b0569d662ebf2e3d9cd988fe8abc40e4cc